frontend/src/store/index.ts

```diff
@@ -0,0 +1,66 @@
+import Vue from 'vue';
+import Vuex from 'vuex';
+import axios from '@/helpers/api.ts'
+import { Game, Player } from '../../types'
+import { get } from 'lodash'
+
+Vue.use(Vuex);
+
+export default new Vuex.Store({
+  state: {
+    game: {} as Game,
+    player: {} as Player,
+  },
+  getters: {
+    apiFunction: (state) => (payload = {}, id?: number) => {
+      return {
+        game: {
+          create: {
+            url: `/games`,
+            method: 'post',
+          },
+          get: {
+            url: `/games/${id}`,
+            method: 'get',
+          },
+        },
+      }
+    },
+    callAPI: (state, getters) => ({ apiAction = {}, payload = {} }) => {
+      return axios()({ ...apiAction, data: payload })
+    },
+  },
+  mutations: {
+    setGame(state, game) {
+      state.game = game
+    },
+    setPlayer(state, player) {
+      state.player = player
+    },
+  },
+  actions: {
+    executeAPI: ({ state, getters, dispatch }, { action, payload }) => {
```

Comment:
I guess it's hart to type this using this code, but that could be done in a separate PR. :)